src/test/net_tests.cpp

```diff
@@ -929,4 +1004,57 @@ BOOST_AUTO_TEST_CASE(node_eviction_test)
     }
 }
 
+BOOST_AUTO_TEST_CASE(initial_advertise_from_version_message)
+{
+    // Pretend that we bound to this port.
+    const uint16_t bind_port = 20001;
+    m_node.args->ForceSetArg("-bind", strprintf("3.4.5.6:%u", bind_port));
+
+    // Our address:port as seen from the peer - 2.3.4.5:20002 (different from the above).
+    in_addr peer_us_addr;
+    peer_us_addr.s_addr = htonl(0x02030405);
+    const CAddress peer_us{CService{peer_us_addr, 20002}, NODE_NETWORK};
+
+    // Create a peer with a routable IPv4 address.
+    in_addr peer_in_addr;
+    peer_in_addr.s_addr = htonl(0x01020304);
+    CNode peer{0,
+               NODE_NETWORK,
+               INVALID_SOCKET,
+               CAddress{CService{peer_in_addr, 8333}, NODE_NETWORK},
+               0,
+               0,
+               CAddress{},
+               std::string{},
+               ConnectionType::OUTBOUND_FULL_RELAY};
+
+    const uint64_t services{NODE_NETWORK | NODE_WITNESS};
+    const int64_t time{0};
+    // Our address, as seen from the peer.
+    const CNetMsgMaker msg_maker{PROTOCOL_VERSION};
+    const auto msg = msg_maker.Make(NetMsgType::VERSION, PROTOCOL_VERSION, services, time, peer_us);
+
+    CDataStream recv_stream{msg.data, SER_NETWORK, PROTOCOL_VERSION};
+
+    // Force CChainState::IsInitialBlockDownload() to return false.
+    // Otherwise PushAddress() isn't called by PeerManager::ProcessMessage().
+    TestChainState& chainstate = *(TestChainState*)&m_node.chainman->ActiveChainstate();
+    chainstate.JumpOutOfIbd();
+
+    m_node.peerman->InitializeNode(&peer);
+
+    std::atomic<bool> interrupt_dummy{false};
+    std::chrono::microseconds time_received_dummy{0};
+
+    m_node.peerman->ProcessMessage(peer, NetMsgType::VERSION, recv_stream, time_received_dummy,
+                                   interrupt_dummy);
```

Comment:
I presume this line will make appveyor etc fail if the test process isn't restarted in between test cases.

Reply:
See

```
Test cases order is shuffled using seed: 193951562
```

https://cirrus-ci.com/task/5953114714931200?command=ci#L3792

Which should make it reproducible

Reply:
Thanks, that helped a lot nailing the issue down. Fixed now.